### PR TITLE
Add Google Cloud Storage handler

### PR DESCRIPTION
**Pull request by davidbrochart**
This only handles anonymous GCS clients for now.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -73,6 +73,8 @@ endif()
 
 find_package(Threads)
 
+find_package(storage_client)
```
Comment (JohanMabille) [inline test/CMakeLists.txt:76]: You may want to add that to the CMakeLists of `xtensor-io` instead, following this [pattern](https://github.com/xtensor-stack/xtensor-io/blob/bab241abb3de8bad7c53f866208d68ebd7affb6d/CMakeLists.txt#L113). This way, the link is automagic for projects using `xtensor-io` if the package is found.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -130,6 +133,7 @@ target_link_libraries(test_xtensor_io_lib
     ${CMAKE_DL_LIBS}
     ${GTEST_BOTH_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
+    storage_client
```
Comment (JohanMabille) [inline test/CMakeLists.txt:136]: This is not required anymore if you move the dependency on GCS to the main CMakeLists.

File: include/xtensor-io/xio_blosc.hpp [hunk 3]
```diff
@@ -34,11 +34,8 @@ namespace xt
         inline xt::svector<T> load_blosc_file(std::istream& stream, bool as_big_endian)
         {
             init_blosc();
-            stream.seekg(0, stream.end);
-            auto compressed_size = static_cast<std::size_t>(stream.tellg());
-            stream.seekg(0, stream.beg);
-            xt::svector<char> compressed_buffer(compressed_size);
-            stream.read(compressed_buffer.data(), (std::streamsize)compressed_size);
+            std::string compressed_buffer{std::istreambuf_iterator<char>{stream}, {}};
+            auto compressed_size = compressed_buffer.size();
```
Comment (JohanMabille) [inline include/xtensor-io/xio_blosc.hpp:38]: Neat!

File: include/xtensor-io/xio_binary.hpp [hunk 4]
```diff
@@ -22,12 +22,10 @@ namespace xt
...
-            stream.seekg(0, stream.end);
-            auto uncompressed_size = static_cast<std::size_t>(stream.tellg());
-            stream.seekg(0, stream.beg);
-            xt::svector<T> uncompressed_buffer(uncompressed_size / sizeof(T));
-            char* buffer = reinterpret_cast<char*>(uncompressed_buffer.data());
-            stream.read(buffer, (std::streamsize)uncompressed_size);
+            std::string buffer{std::istreambuf_iterator<char>{stream}, {}};
+            std::size_t uncompressed_size = buffer.size() / sizeof(T);
+            xt::svector<T> uncompressed_buffer(uncompressed_size);
+            std::copy((const T*)(buffer.data()), (const T*)(buffer.data()) + uncompressed_size, uncompressed_buffer.begin());
```
Comment (davidbrochart) [inline include/xtensor-io/xio_binary.hpp:28]: This creates two copies, one in `buffer` and one in `uncompressed_buffer`, but I could not make `istreambuf_iterator` work directly with `xt::svector`.
Any idea?

Comment (JohanMabille) [reply include/xtensor-io/xio_binary.hpp:28]: Maybe something like:
```cpp
auto it = std::istreambuf_iterator<char>(stream);
std::transform(std::istreambuf_iterator<char>(stream), (std::istreambuf_iterator<char>()), std::back_inserter(uncompressed_buffer), [&it](char c) 
{
    char tmp[sizeof(T)];
    tmp[0] = c;
    std::copy_n(++it, sizeof(T)-1, tmp+1);
    T* res = reinterpret_cast<T*>(tmp);
    return res;
});
```

But I'm not really sure that this is really readable ;)

Comment (davidbrochart) [reply include/xtensor-io/xio_binary.hpp:28]: Ok, let's keep the copy for now then.

**Pull request merged**